server/channels/store/sqlstore/post_store.go

```diff
@@ -1043,20 +1057,39 @@ func (s *SqlPostStore) PermanentDeleteByChannel(channelId string) (err error) {
 	}
 	defer finalizeTransactionX(transaction, &err)
 
-	results := []postIds{}
-	err = transaction.Select(&results, "SELECT Id, RootId, UserId FROM Posts WHERE ChannelId = ?", channelId)
-	if err != nil {
-		return errors.Wrapf(err, "failed to fetch Posts with channelId=%s", channelId)
-	}
+	id := ""
+	for {
+		ids := []string{}
+		err = transaction.Select(&ids, "SELECT Id FROM Posts WHERE ChannelId = ? AND Id > ? ORDER BY Id ASC LIMIT 1000", channelId, id)
```

Comment:
On a deeper look, I didn't realize that there isn't an index with channelId+id. So the order might be slightly costly.

```
[bigdb] # EXPLAIN ANALYZE SELECT Id FROM Posts WHERE ChannelId = 'bjmg1did3pfiufccedjaoht6hw' AND Id > 'hmdg8px5pjnsmjcetskec557wr' ORDER BY Id ASC LIMIT 1000;
                                                                         QUERY PLAN
------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=7450.50..7452.94 rows=974 width=27) (actual time=78.972..79.271 rows=1000 loops=1)
   ->  Sort  (cost=7450.50..7452.94 rows=974 width=27) (actual time=78.969..79.104 rows=1000 loops=1)
         Sort Key: id
         Sort Method: top-N heapsort  Memory: 179kB
         ->  Bitmap Heap Scan on posts  (cost=71.13..7402.15 rows=974 width=27) (actual time=17.877..67.433 rows=2775 loops=1)
               Recheck Cond: ((channelid)::text = 'bjmg1did3pfiufccedjaoht6hw'::text)
               Filter: ((id)::text > 'hmdg8px5pjnsmjcetskec557wr'::text)
               Rows Removed by Filter: 2707
               Heap Blocks: exact=410
               ->  Bitmap Index Scan on idx_posts_channel_id_update_at  (cost=0.00..70.89 rows=1910 width=0) (actual time=17.566..17.567 rows=5482 loops=1)
                     Index Cond: ((channelid)::text = 'bjmg1did3pfiufccedjaoht6hw'::text)
 Planning Time: 2.431 ms
 Execution Time: 79.508 ms
```

Let's change the page size to 500 to be on the safe side.